### PR TITLE
Fix location search bug

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of how you validated changes
- Fixes https://github.com/aws-amplify/amplify-ui/issues/1797
- Upgraes maplibre-gl-geocoder and maplibre-gl-js-amplify  - https://github.com/maplibre/maplibre-gl-geocoder/commit/8a76227be4c401505430c5648bf77a390eb7f9c4

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ioanabrooks** [comment]
LGTM. Is this a fix for https://github.com/aws-amplify/amplify-ui/issues/1797? If so please update the description to indicate that with the fixes prefix.

**Comment by thaddmt** [comment]
@ioanabrooks haha yes it is thanks! You found it before I pinged it out :)

**Review by ioanabrooks — DISMISSED** [review]

**Review by wlee221 — APPROVED** [review]

File: packages/react/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "lodash": "4.17.21",
     "mapbox-gl": "1.13.1",
     "maplibre-gl": "2.1.9",
-    "maplibre-gl-js-amplify": "2.0.2",
+    "maplibre-gl-js-amplify": "^2.0.4",
```
Comment (zchenwei) [inline packages/react/package.json:70]: Is there a reason for not pinning the dependency? We do that for other direct dependencies

Comment (thaddmt) [reply packages/react/package.json:70]: Good catch, that was a mistake!

**Review by zchenwei — DISMISSED** [review]
LGTM 🚢

**Review by ioanabrooks — DISMISSED** [review]

**Review by ioanabrooks — DISMISSED** [review]

**Review by calebpollman — APPROVED** [review]
LGTM 🚢

**Review by wlee221 — APPROVED** [review]

**Review by slaymance — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Pull request merged**